dynamiqs/utils/operators.py

```diff
@@ -330,6 +334,94 @@ def squeeze(
     return torch.matrix_exp(0.5 * (z.conj() * a2 - z * a2.mH))
 
 
+def quadrature(
+    dim: int,
+    phi: float,
+    *,
+    dtype: torch.complex64 | torch.complex128 | None = None,
+    device: str | torch.device | None = None,
+) -> Tensor:
+    r"""Returns the quadrature operator of phase angle $\phi$.
+
+    It is defined by $x(\phi) = (e^{i\phi} a + e^{-i\phi} a^\dag) / 2$, where $a$ and
+    $a^\dag$ are the annihilation and creation operators respectively.
+
+    Args:
+        dim: Dimension of the Hilbert space.
+        phi: Phase angle.
+        dtype: Complex data type of the returned tensor.
+        device: Device of the returned tensor.
+
+    Returns:
+        _(dim, dim)_ Quadrature operator.
+
+    Examples:
+        >>> from math import pi
+        >>> dq.quadrature(3, 0.0)
+        tensor([[0.000+0.j, 0.707+0.j, 0.000+0.j],
+                [0.707+0.j, 0.000+0.j, 1.000+0.j],
+                [0.000+0.j, 1.000+0.j, 0.000+0.j]])
+        >>> dq.quadrature(3, pi / 2)
+        tensor([[0.0000e+00+0.0000j, 4.3298e-17+0.7071j, 0.0000e+00+0.0000j],
+                [4.3298e-17-0.7071j, 0.0000e+00+0.0000j, 6.1232e-17+1.0000j],
+                [0.0000e+00+0.0000j, 6.1232e-17-1.0000j, 0.0000e+00+0.0000j]])
```

Comment:
I'm pretty sure doctest are not passing here (because it's precision 3 usually).